### ISSUE TITLE
cli/sql: time-related types often display extraneous info

### ISSUE THREAD

**Issue by solongordon** [cockroachdb/cockroach#21390]
`pq` represents all time-related types (`DATE`, `TIME`, `TIMESTAMP`, `TIMESTAMPTZ`) using Go's `time.Time`. As a result, `cockroach sql` often uses too much precision when it displays these types. This is only a display bug but it can be confusing for users.

The most egregious example is `TIME`, where we display an irrelevant date and timezone.
```
root@:26257/> select '02:03:04'::TIME;
+---------------------------+
|     '02:03:04'::TIME      |
+---------------------------+
| 0000-01-01 02:03:04+00:00 |
+---------------------------+
```
But `DATE` and `TIMESTAMP` are also somewhat misleading.
```
root@:26257/> select '2018-01-10'::DATE;  -- should not include time
+---------------------------+
|    '2018-01-10'::DATE     |
+---------------------------+
| 2018-01-10 00:00:00+00:00 |
+---------------------------+

root@:26257/> select '2018-01-10 02:03:04'::TIMESTAMP;  -- should not include timezone
+----------------------------------+
| '2018-01-10 02:03:04'::TIMESTAMP |
+----------------------------------+
| 2018-01-10 02:03:04+00:00        |
+----------------------------------+
```

**Comment by otan** [comment]
related to https://github.com/cockroachdb/cockroach/issues/44548, i think this results in real bugs when using things such as client dumps.

**Comment by knz** [comment]
Fixed in #62310.
